frontend/src/static/js/v_summary.js

```diff
@@ -492,13 +492,52 @@ window.vSummary = {
       user.dailyCommits.forEach((commit) => {
         const { date } = commit;
         if (date >= sinceDate && date <= untilDate) {
-          user.commits.push(commit);
+          const filteredCommit = JSON.parse(JSON.stringify(commit));
+          if (this.filterBreakdown) {
+            this.filterCommitByCheckedFileTypes(filteredCommit);
+          }
+          if (filteredCommit.commitResults.length > 0) {
+            user.commits.push(filteredCommit);
+          }
         }
       });
 
       return null;
     },
 
+    filterCommitByCheckedFileTypes(commit) {
+      const filteredCommitResults = commit.commitResults.map((result) => {
+        const filteredFileTypes = this.getFilteredFileTypes(result);
+        this.updateCommitResultWithFileTypes(result, filteredFileTypes);
+        return result;
+      }).filter((result) => Object.values(result.fileTypesAndContributionMap).length > 0);
+
+      commit.insertions = filteredCommitResults.reduce((acc, result) => acc + result.insertions, 0);
+      commit.deletions = filteredCommitResults.reduce((acc, result) => acc + result.deletions, 0);
+      commit.commitResults = filteredCommitResults;
+    },
+
+    getFilteredFileTypes(commitResult) {
+      return Object.keys(commitResult.fileTypesAndContributionMap)
+          .filter((fileType) => this.isFileTypeIncluded(fileType))
```

Comment:
Can rename `isFileTypeIncluded` to `isFileTypeChecked` to be clearer.

Also, I think you can just rewrite `.filter((fileType) => this.isFileTypeIncluded(fileType))` as `.filter(this.isFileTypeIncluded)`

Reply:
Done. Thanks for the feedback.